server/src/main/java/org/elasticsearch/action/search/SearchTransportService.java

```diff
@@ -400,14 +400,31 @@ public static void registerRequestHandler(TransportService transportService, Sea
         );
 
         // TODO: remove this handler once the lowest compatible version stops using it
+        // this handler exists for BwC purposes only, we don't need the original indices to free the context
         transportService.registerRequestHandler(FREE_CONTEXT_ACTION_NAME, freeContextExecutor, in -> {
             var res = new ScrollFreeContextRequest(in);
             // this handler exists for BwC purposes only, we don't need the original indices to free the context
             OriginalIndices.readOriginalIndices(in);
             return res;
         }, freeContextHandler);
-        TransportActionProxy.registerProxyAction(transportService, FREE_CONTEXT_ACTION_NAME, false, SearchFreeContextResponse::readFrom);
-
+        // BwC handler that translates FREE_CONTEXT_ACTION_NAME into FREE_CONTEXT_SCROLL_ACTION_NAME same way we use
+        // FREE_CONTEXT_SCROLL_ACTION_NAME throughout on local nodes
+        transportService.registerRequestHandler(
```

Comment:
I'm not too proud of exposing this infrastructure just for a one-off fix to be honest. But then again, hiding it behind some utility method and adding a single-use abstraction for the action-to-action translation seemed even worse. This solution is the minimum amount of code needed as far as I can tell which is quite helpful.